src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsstrengthsbasedneedsassessmentsapi/service/AssessmentVersionService.kt

```diff
@@ -18,30 +20,23 @@ class AssessmentVersionService(
   val assessmentVersionRepository: AssessmentVersionRepository,
   val dataMappingService: DataMappingService,
 ) {
-  fun create(tag: String, assessment: Assessment): AssessmentVersion {
+  fun sameOrCloneFromPrevious(tag: String, assessment: Assessment): AssessmentVersion {
```

Comment:
we could retain the `create` verb here for example `createOrCloneFromPrevious`

Reply:
🤔 clone and create are the same thing in this context... maybe "createOrReuse"?

Reply:
or "createOrGetPrevious"?

Reply:
true and makes sense, maybe `createNewOrCloneFromPrevious`, but I'm also happy with either of the above suggestions 👍

Reply:
ah I see what you mean now. There's actually 3 scenarios here:
- create new (no previous version exists)
- clone from previous (previous version is older than 1 day)
- get previous (previous version was created today)

so I guess `createNewOrGetPreviousOrCloneFromPrevious`

Reply:
Think I've settled on `getPreviousOrCreate` :D describes all scenarios while not being ridiculously long